src/GafferAppleseedModule/GafferAppleseedModule.cpp

```diff
@@ -50,12 +50,49 @@
 using namespace boost::python;
 using namespace GafferAppleseed;
 
+namespace {
+
+class AppleseedLightSerialiser : public GafferBindings::NodeSerialiser
+{
+
+	std::string postConstructor( const Gaffer::GraphComponent *graphComponent, const std::string &identifier, const GafferBindings::Serialisation &serialisation ) const override
+	{
+		const AppleseedLight *light = static_cast<const AppleseedLight *>( graphComponent );
+
+		// \todo - Remove this once old scripts have been converted
+		// Before we start serialization, clean up any old scripts that might have dynamic parameters on lights
+		// ( Now we create the parameters with a loadShader after the constructor, so they don't need to be dynamic )
+		for( auto i : light->parametersPlug()->children() )
+		{
+			Gaffer::Plug *p = IECore::runTimeCast< Gaffer::Plug >( i.get() );
+			if( p )
+			{
+				p->setFlags( Gaffer::Plug::Dynamic, false );
+			}
+		}
+
+		// \todo - Accessing this plug by explicit name is ugly, but the accessor is private.
```

Comment:
It seems a pity that we're having to make serialisers for each light subclass individually, rather than just having one LightSerialiser that works for all of them. Is this the only reason for that, or are there others too? In the interests of moving things closer to where we want them to be, I'd be happy to see this hack included in a single central LightSerialiser class if we can make that work.

I think another issue would be that the Light base class doesn't have a `loadShader()` method, but we can look for it is a python attribute in the same way as the `ValuePlugSerialiser` [looks for `getValue()`](https://github.com/GafferHQ/gaffer/blob/master/src/GafferBindings/ValuePlugBinding.cpp#L141).

Reply:
This would also require standardizing on a way to get the shader name.  Currently we have:

public: OSLLight::shaderNamePlug() 
private: ArnoldLight::shaderNamePlug() 
private: AppleseedLight::modelPlug()

If you want to include this in the base class, should we just standardize on lights having a loadShader and shaderNamePlug(), and make these virtual in the base class?  ( I kinda feel like shaderName could just be a member variable instead of a plug, but you probably want to keep the possibility of driving it dynamically somehow, since you didn't like the idea of passing it to the constructor. )

Reply:
For now I was thinking we could just always access the plug by the `getChild<StringPlug>( name )` method you used here, looking for the three names in question. That'll be ugly for sure, but it gives us a single serialiser, and a single place to put a big comment saying that we should standardise.

I think the consideration of plug vs member should be done when refactoring the Light base class to adopt the OSLLight's "defer to internal Shader node" approach across the board. I think that's out of the scope of this PR though - we've already got the serialisation to where we want it to be, the rest can wait...